### DISCUSSION TITLE
Disconnect Wallet button does not really disconnect the wallet

### DISCUSSION THREAD

**Discussion by perdiweb3** [rainbow-me/rainbowkit#693] [Q&A]
Hello.

I am doing some stuff with the library, using Windows 11 and Google Chrome, and I realized something. When my site is connected to the wallet, the library offers you the possibility to disconnect the wallet.

<img width="235" alt="image" src="https://user-images.githubusercontent.com/107559416/187172469-7bbf5f3c-f44b-4442-bc5a-75f0bfaa5e22.png">

However, this action does not really disconnect the wallet from the site because the wallet, in my case Metamask, is still connected and signing transactions is still available.

My question is: **Is this a feature of the library or it exists a way to completely disconnect the wallet from the site?**

Thank you very much for your time and attention!

**Comment by aalmada** [comment]
I've already implemented wallet integration using the injected `window.ethereum`, using wagmi and using RainbowKit. Disconnect doesn't work on any of those for the wallets I tested (Metamask and Coinbase). After "disconnecting", if I refresh the page, I find it connected again. I don't understand why libraries provide this if it doesn't work.
I have my own connect button and I don't provide disconnect to the user. I added a tooltip telling the user to open their wallet and disconnect from there.

**Comment by DanielSinclair** [comment]
MetaMask and other wallets don't currently support programmatic disconnect. Wagmi has a feature called `shimDisconnect` that tries to mock this behavior by "hiding" that the wallet provider is available at `window.ethereum`, and the support there has improved in recent months.
